### PR TITLE
Changed search issues api - adding filtering issues by issue numbers

### PR DESCRIPTION
**Pull request by alubenskyi**

### CODE REVIEW

File: spec/openapi.yaml [hunk 1]
```diff
@@ -3927,7 +3927,7 @@ paths:
       description: >-
         Returns a list of issues matching specified filter options. You can
         filter based on the date issues were created, target languages, strings,
-        issue types, states, and which user opened the issue. Unless otherwise
+        issue types, states, issue numbers, and which user opened the issue. Unless otherwise
```
Comment (hgronberg-smartling) [inline spec/openapi.yaml]: Does issue number mean and issue ID, or the number of issues on a string or in a project? If we have this defined elsewhere, no change is needed, but just reading this by itself I don't know what issue numbers would be.

File: spec/openapi.yaml [hunk 2]
```diff
@@ -3990,7 +3990,7 @@ paths:
       description: >-
         Returns a count of issues matching specified filter options. You can
         filter based on the date issues were created, strings, target languages,
-        issue types, states, and which user opened the issue. All parameters are
+        issue types, states, issue numbers, and which user opened the issue. All parameters are
```
Comment (hgronberg-smartling) [inline spec/openapi.yaml]: Same comment as above

File: spec/openapi.yaml [hunk 3]
```diff
@@ -10151,6 +10151,16 @@ components:
           description: User identifier who created issue
           example: 661801f19693
           type: string
+        issueNumbers:
```
Comment (hgronberg-smartling) [inline spec/openapi.yaml]: Same comment as above - issueNumbers sounds like a number of issues associated with some object. If it is the unique ID of an issue or something like that, it should be called issueID or something along those lines?

File: spec/openapi.yaml (hunk 1 above)
Comment (alubenskyi) [reply spec/openapi.yaml]: Issue number - sequence number within an account.

Comment (DmitryBochorishvili) [reply spec/openapi.yaml]: It sounds like an internal issue identifier. Do we really need to expose such internal stuff in public api?

Comment (alubenskyi) [reply spec/openapi.yaml]: I think yes so as this number is displayed on UI and we already have filter by issue number on "Issues" page

Comment (AlexArcher) [reply spec/openapi.yaml]: Hi guys,

Right now we have 
issue_id - internal issue identifier, which is unique for the whole DB, we do not expose it to users
issue_uid - unique issue identifier which we show to users
they need to use it to open/close/update issue
and issue_number was requested by users as an additional reference (alias)

For example:
I'm saying: Bob, can you review issue #32
and Bob will be able to find it and review.
Uid is not user-friendly.

As @alubenskyi  said issue number is unique only for a specific account.
If I create a new account and create issue then the number will be  #1 
But other accounts may also have an issue with #1

If you have any questions - reach us in a Slack and let's make a conference call.

Comment (DmitryBochorishvili) [reply spec/openapi.yaml]: Thanks for the thorough explanation, Alex. Sounds like `issue reference number` might be a good name for this new field. @hgronberg-smartling, what do you think whether that proposed name may resolve confusion around the field?

Comment (hgronberg-smartling) [reply spec/openapi.yaml]: Hi guys, thanks for the explanation - this makes a lot of sense now. It looks like we name this filter "Issue Number" in the UI (Reports page), so it is probably best to keep it consistent here.

**Pull request merged**